vyper/optimizer.py

```diff
@@ -87,17 +171,47 @@ def optimize(node):
         if get_int_at(argz, 0, True) > get_int_at(argz, 1, True):
             raise Exception("Clamp always fails")
         else:
-            return LLLnode("clample", [argz[1], argz[2]], node.typ, node.location, node.pos, node.annotation, add_gas_estimate=node.add_gas_estimate, valency=node.valency)
+            return LLLnode(
+                "clample",
+                [argz[1], argz[2]],
+                node.typ,
+                node.location,
+                node.pos,
+                node.annotation,
+                add_gas_estimate=node.add_gas_estimate,
+                valency=node.valency,
+            )
     elif node.value == "clamp_nonzero" and int_at(argz, 0):
         if get_int_at(argz, 0) != 0:
-            return LLLnode(argz[0].value, [], node.typ, node.location, node.pos, node.annotation, add_gas_estimate=node.add_gas_estimate, valency=node.valency)
+            return LLLnode(
+                argz[0].value,
+                [],
+                node.typ,
+                node.location,
+                node.pos,
+                node.annotation,
+                add_gas_estimate=node.add_gas_estimate,
+                valency=node.valency,
+            )
         else:
             raise Exception("Clamp always fails")
     # [eq, x, 0] is the same as [iszero, x].
     elif node.value == 'eq' and int_at(argz, 1) and argz[1].value == 0:
-        return LLLnode('iszero', [argz[0]], node.typ, node.location, node.pos, node.annotation, add_gas_estimate=node.add_gas_estimate, valency=node.valency)
-    # Turns out this is actually not such a good optimization after all
-    elif node.value == "with" and int_at(argz, 1) and not search_for_set(argz[2], argz[0].value) and False:
+        return LLLnode(
+            'iszero',
+            [argz[0]],
+            node.typ,
+            node.location,
+            node.pos,
+            node.annotation,
+            add_gas_estimate=node.add_gas_estimate,
+            valency=node.valency,
+        )
+    elif _is_with_without_set(node, argz):
+        # TODO: This block is currently unreachable due to
+        # `_is_with_without_set` unconditionally returning `False` this appears
+        # to be because this "is actually not such a good optimization after
+        # all" accordiing to previous comment.
```

Comment:
@jacqueswww, should evalulate this

Reply:
I created an issue https://github.com/ethereum/vyper/issues/1306, I suspect we can just drop the optimisation.